### PR TITLE
feat(ingest): redact configs reported in ingestion_run_summary

### PR DESCRIPTION
**Pull request by hsheth2**
This approach is somewhat of a hack. We don't have access to the actual config classes here, so tagging fields in the underlying config models is not a viable approach.

## Checklist

- [ ] The PR conforms to DataHub's [Contributing Guideline](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable). If a new feature has been added a Usage Guide has been added for the same.
- [ ] For any breaking change/potential downtime/deprecation/big changes an entry has been made in [Updating DataHub](https://github.com/datahub-project/datahub/blob/master/docs/how/updating-datahub.md)

### CODE REVIEW

File: metadata-ingestion/src/datahub/configuration/common.py [hunk 1]
```diff
@@ -15,6 +15,58 @@
...
+        return None
+    elif isinstance(value, bool):
+        # We don't have any sensitive boolean fields.
+        return value
+    elif isinstance(value, list) and not value:
+        # Empty states are fine.
+        return []
+    elif isinstance(value, dict) and not value:
+        return {}
+    else:
```
Comment (treff7es) [inline metadata-ingestion/src/datahub/configuration/common.py:54]: Was it a conscious decision not to check number types?
I don't think it is an issue here, I was just wondering.

File: metadata-ingestion/src/datahub/ingestion/reporting/datahub_ingestion_run_summary_provider.py [hunk 2]
```diff
@@ -149,7 +154,7 @@ def _get_recipe_to_report(self, ctx: PipelineContext) -> str:
         if not self.report_recipe or not ctx.pipeline_config._raw_dict:
             return ""
         else:
-            return json.dumps(ctx.pipeline_config._raw_dict)
+            return json.dumps(redact_raw_config(ctx.pipeline_config._raw_dict))
```
Comment (treff7es) [inline metadata-ingestion/src/datahub/ingestion/reporting/datahub_ingestion_run_summary_provider.py:157]: Is this the only place we should redact the password fields, or was this PR about to redact the password from here?

**Review by treff7es — APPROVED** [review]
lgtm

File: metadata-ingestion/src/datahub/configuration/common.py (hunk 1 above)
Comment (hsheth2) [reply metadata-ingestion/src/datahub/configuration/common.py:54]: I've only not redacted bools, env refs, and empty lists/dicts. I figured that a number could be an account ID or something and hence should be redacted in general.

File: metadata-ingestion/src/datahub/ingestion/reporting/datahub_ingestion_run_summary_provider.py (hunk 2 above)
Comment (hsheth2) [reply metadata-ingestion/src/datahub/ingestion/reporting/datahub_ingestion_run_summary_provider.py:157]: This PR was to redact the password showing up there, but will be useful if we ever need to write config to logs in the future as well.

**Pull request merged**